### PR TITLE
Use unified diff for recipe solutions

### LINKED ISSUE #107: Recipe edit solutions use unified diff format
On https://esmvalgroup.github.io/ESMValTool_Tutorial/04-recipe/index.html#do-your-first-edits some solutions are presented as changes that need to be made to [data/recipe_example.yml](https://github.com/ESMValGroup/ESMValTool_Tutorial/blob/master/data/recipe_example.yml) . For me it took some time to understand that lines with `XX` should be added.

We could use the unified diff format 

```diff
--- data/recipe_example.yml	2020-07-31 14:43:48.976660742 +0200
+++ data/recipe_example_tos.yml	2020-08-03 14:11:33.721000000 +0200
@@ -20,12 +20,16 @@
     - ukesm
 
 datasets:
-  - {dataset: HadGEM2-ES, project: CMIP5, exp: historical, mip: Omon, ensemble: r1i1p1, start_year: 1859, end_year: 2005}
+  - {dataset: HadGEM2-AO, project: CMIP5, exp: historical, mip: Omon, ensemble: r1i1p1, start_year: 1901, end_year: 2000}
+  - {dataset: HadGEM2-CC, project: CMIP5, exp: historical, mip: Omon, ensemble: r1i1p1, start_year: 1901, end_year: 2000}
+  - {dataset: HadGEM2-ES, project: CMIP5, exp: historical, mip: Omon, ensemble: r1i1p1, start_year: 1901, end_year: 2000}
 
 preprocessors:
   prep_timeseries: # For 0D fields
     annual_statistics:
       operator: mean
+    area_statistics:
+      operator: mean
 
 diagnostics:
   # --------------------------------------------------
@@ -35,7 +39,7 @@
     description: simple_time_series
     variables:
       timeseries_variable:
-        short_name: thetaoga
+        short_name: tos
         preprocessor: prep_timeseries
     scripts:
       timeseries_diag:
```

This to me is much easier to read.

### PR DESCRIPTION
**Pull request by sverhoeven**
Fixed #107

# Pull Request checklist

We appreciate your time and effort to improve the tutorial. Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution.

* * *

## Before you start

- [x] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
- [x] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss your idea. This allows your contributions to be incorporated into the tutorial.

## Tasks

- [ ] Give this pull request a descriptive title.
- [x] If you are contributing to existing lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [x] If you are making a new episode, please make sure the content conforms to the `Lesson organization` and `Lesson formatting` sections in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] Preferably Codacy checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why.
- [x] Preview changes on your machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. Please see the `Previewing your changes locally` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) for installation instructions.
- [ ] All code instructions have been tested.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes #107

### CODE REVIEW

File: data/recipe_example_tas.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# ESMValTool
+# recipe_example.yml
```
[the remark this answers is not in the document]
Comment (sverhoeven) [reply data/recipe_example_tas.yml:2]: Yes, I use the solution recipe files for `diff -u orig solution`, but also to check in CI they should be used see #110

File: _episodes/04-recipe.md [hunk 2]
```diff
@@ -635,58 +635,110 @@ The snippets for the edits can be found below:
...
-> 28          operator: mean
-> XX        area_statistics:
-> XX          operator: mean
->  ...
-> 38            short_name: ts
-> 39            preprocessor: prep_timeseries
+> ```diff
+> --- data/recipe_example.yml	2020-07-31 14:43:48.976660742 +0200
+> +++ data/recipe_example_ts.yml	2020-08-03 14:57:45.962391097 +0200
+> @@ -20,12 +20,14 @@
```
[the remark this answers is not in the document]
Comment (sverhoeven) [reply _episodes/04-recipe.md:641]: I can remove the timestamps, it still renders OK. 
I would like to keep the filename as for me the filename makes clear what is the original and the target.

File: _episodes/04-recipe.md [hunk 3]
```diff
@@ -635,58 +635,110 @@ The snippets for the edits can be found below:
...
+>  preprocessors:
+>    prep_timeseries: # For 0D fields
+>      annual_statistics:
+>        operator: mean
+> +    area_statistics:
+> +      operator: mean
+>
+>  diagnostics:
+>    # --------------------------------------------------
+> @@ -35,7 +37,7 @@
```
[the remark this answers is not in the document]
Comment (sverhoeven) [reply _episodes/04-recipe.md:657]: I can reduce the number of line around the changed lines, but I would like to see the parent key to make the context clear. The default is 3. If I reduce it to 2 or 1 I loose the parent keys for the preprocessors.

File: _episodes/04-recipe.md [hunk 4]
```diff
@@ -635,58 +635,116 @@ The snippets for the edits can be found below:
 
 > ## Land surface average temperature
 >
```
Comment (Peter9192) [inline _episodes/04-recipe.md:637]: Maybe we can help the students understand the diff format a bit better then? 
```suggestion
>
> In the `diff` file below you will see the changes we have made to the file. The top 2 lines are the filenames and the lines like @@ -20,12 +20,14 @@ indicate the line numbers in the original and modified file, respectively. For more info on this format, see [here](https://en.wikipedia.org/wiki/Diff#Unified_format)
>
```

Comment (sverhoeven) [reply _episodes/04-recipe.md:637]: Great suggestion

**Pull request merged**